frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx

```diff
@@ -100,6 +101,21 @@ const ManageInferenceServiceModal: React.FC<ManageInferenceServiceModalProps> =
     return isConnectionValid;
   };
 
+  React.useEffect(() => {
+    if (createData.name && createData.name !== inferenceServiceNameDesc.name) {
+      setInferenceServiceNameDesc('name', createData.name);
+    }
+    setIsLoading(false);
+  }, [inferenceServiceNameDesc.name, createData.name, setInferenceServiceNameDesc]);
+
+  React.useEffect(() => {
+    if (modelDeployPrefillInfo) {
+      const isDataReady = !!inferenceServiceNameDesc.name && (!projectContext || connectionsLoaded);
+
+      setIsLoading(!isDataReady);
+    }
+  }, [inferenceServiceNameDesc.name, projectContext, connectionsLoaded, modelDeployPrefillInfo]);
```

Comment:
It would be great to not have `isLoading` be state if we can avoid it. Can we derive it at render time as a regular variable and prevent needing this useEffect?

```ts
const isDataReady = !modelDeployPrefillInfo || (!!inferenceServiceNameDesc.name && (!projectContext || connectionsLoaded));
const isLoading = !isDataReady;
```